File: 3412-permutation-difference-between-two-strings/3412-permutation-difference-between-two-strings.java
class Solution {
    public int findPermutationDifference(String s, String t) {
        HashMap<Character,Integer> hs = new HashMap<>();
        int ans =0;
        for(int i=0;i<t.length();i++){
            hs.put(t.charAt(i),i);
        }
        for(int i=0;i<s.length();i++){
            int tk  = hs.get(s.charAt(i));
            int g = Math.abs(tk-i);
            ans+=g;
        }

        return ans;
    }
}

File: 3412-permutation-difference-between-two-strings/3412-permutation-difference-between-two-strings.cpp
class Solution {
public:
    int findPermutationDifference(string s, string t) {

       int diff = 0;
       for(int i = 0; i<s.size(); i++){
        int sum = 0;
          for(int j = 0; j<t.size(); j++){
              if(s[i] == t[j]){
                sum = abs(i-j);
              }
          }
       diff +=sum;
       } 
    return diff;
    }
};